builtin/credential/cert/backend_test.go

```diff
@@ -76,37 +77,69 @@ func connectionState(t *testing.T, serverCAPath, serverCertPath, serverKeyPath,
 	// Start the server.
 	list, err := tls.Listen("tcp", "127.0.0.1:0", listenConf)
 	if err != nil {
-		t.Fatal(err)
+		return tls.ConnectionState{}, err
 	}
 	defer list.Close()
 
+	// Accept connections.
+	serverErrors := make(chan error, 1)
+	connState := make(chan tls.ConnectionState)
+	go func() {
+		defer close(connState)
+		serverConn, err := list.Accept()
+		serverErrors <- err
```

Comment:
It feels odd to me to write nil errors onto the channel (here and elsewhere). It's not that you're not handling it right, but is there a particular reason you chose to do this?